spec/src/main/asciidoc/ServerPages.adoc

```diff
@@ -9307,21 +9154,12 @@ are very specific to the scripting language used in the page. This is
 especially complex since scriptlets `are language` fragments, not
 complete language statements.
 
-==== [[a2605]]Valid JSP Page
+==== Valid JSP Page
 
 A JSP page is valid for a Java Platform if and
-only if the JSP page implementation class defined by
-link:jsp.html#a2611[See The transformations described in this
-chapter need not be performed literally. An implementation may implement
-things differently to provide better performance, lower memory
-footprint, or other implementation attributes..]
-link:jsp.html#a2612[See Structure of the Java Programming
-Language Class.] (after applying all include directives), together with
-any other classes defined by the JSP container, is a valid program for
-the given Java Platform, and if it passes the validation methods for all
-the tag libraries associated with the JSP page.
```

Comment:
Is this detail you cleaned up (removed?) going to show up somewhere else in the document?
Or is it considered noise / irrelevant now?

Reply:
With my current change the `Valid JSP Page` Section reads:

>  A JSP page is valid for a Java Platform if and only if the JSP page implementation class defined by
> `Structure of the Java Programming Language Class` (after applying all include directives), together with
> any other classes defined by the JSP container, is a valid program for the given Java Platform, and if it
> passes the validation methods for all the tag libraries associated with the JSP page.

Previous to my changes it read:

> A JSP page is valid for a Java Platform if and only if the JSP page implementation class defined by `See
> The transformations described in this chapter need not be performed literally. An implementation may
> implement things differently to provide better performance, lower memory footprint, or other
> implementation attributes.. See Structure of the Java Programming Language Class.` (after applying all
> include directives), together with any other classes defined by the JSP container, is a valid program for
> the given Java Platform, and if it passes the validation methods for all the tag libraries associated with
> the JSP page.

The highlighted part should be a link to the table `Structure of the Java Programming Language Class`

The JSP 2.3 specification states the following:

> A JSP page is valid for a Java Platform if and only if the JSP page implementation
> class defined by Table JSP.9-1 (after applying all include directives), together
> with any other classes defined by the JSP container, is a valid program for the given
> Java Platform, and if it passes the validation methods for all the tag libraries associated
> with the JSP page.

The table referenced: `Table JSP.9-1 Structure of the JavaProgramming Language Class`

The link was formatted in this project as : `link:jsp.html#a2611[See The transformations described in this
chapter need not be performed literally. An implementation may implement
things differently to provide better performance, lower memory
footprint, or other implementation attributes..]`

Looking for that anchor I found:

> ==== Implementation Flexibility
> 
> The transformations described in this chapter
> need not be performed literally. An implementation may implement things
> differently to provide better performance, lower memory footprint, or
> other implementation attributes.[[a2611]]
> 
> _[[a2612]]Structure of the Java Programming Language Class_
> 
> [cols="25,75"]

So I just added a proper anchor for the table and removed the fluff in the link. However as you can see the information that I removed is still in the correct section `Implementation Flexibility` but was just fluff in the link since all we needed as far as I can tell is a link to the table.

```
==== Valid JSP Page

A JSP page is valid for a Java Platform if and
only if the JSP page implementation class defined by <<_Structure_of_the_Java_Programming_Language_Class>> (after applying all include directives), together with any other classes defined by the JSP container, is a valid program for the given Java Platform, and if it passes the validation methods for all the tag libraries associated with the JSP page.
```
```
[[_Structure_of_the_Java_Programming_Language_Class]]
.Structure of the Java Programming Language Class
[cols="25,75"]
```

Reply:
Gotcha.
Thanks.